File: src/services/sessions.ts
import { invoke } from "@tauri-apps/api/core";
import type { SessionMeta, SessionDetail } from "@/types/session";

export async function listSessions(
  limit: number,
  offset: number,
): Promise<SessionMeta[]> {
  return invoke<SessionMeta[]>("list_sessions", { limit, offset });
}

export async function getSessionDetail(
  sessionId: string,
): Promise<SessionDetail> {
  return invoke<SessionDetail>("get_session_detail", {
    sessionId,
  });
}

export async function renameSession(
  sessionId: string,
  name: string,
): Promise<void> {
  return invoke("rename_session", { sessionId, name });
}

export async function deleteSessionFile(
  filePath: string,
): Promise<void> {
  return invoke("delete_session_file", { filePath });
}

File: src-tauri/src/commands/sessions.rs
use crate::services::{session_detail, session_parser, session_types::*};

#[tauri::command]
pub fn list_sessions(
    limit: usize,
    offset: usize,
) -> Result<Vec<SessionMeta>, String> {
    session_parser::list_sessions(limit, offset)
}

#[tauri::command]
pub fn get_session_detail(session_id: String) -> Result<SessionDetail, String> {
    session_detail::get_detail(&session_id)
}

#[tauri::command]
pub fn rename_session(session_id: String, name: String) -> Result<(), String> {
    session_parser::save_session_name(&session_id, &name)
}

#[tauri::command]
pub fn delete_session_file(file_path: String) -> Result<(), String> {
    session_parser::delete_session(&file_path)
}
